Modules/Filtering/ImageGrid/test/itkChangeInformationImageFilterGTest.cxx

```diff
@@ -0,0 +1,57 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+// First include the header file to be tested:
+#include "itkChangeInformationImageFilter.h"
+#include "itkImage.h"
+#include <gtest/gtest.h>
+
+
+// Checks the properties of a newly created ChangeInformationImageFilter.
+TEST(ChangeInformationImageFilter, CheckNew)
+{
+  const auto check = [](const auto & newChangeInformationImageFilter) {
+    ASSERT_NE(newChangeInformationImageFilter, nullptr);
+
+    EXPECT_EQ(newChangeInformationImageFilter->GetReferenceImage(), nullptr);
+
+    EXPECT_FALSE(newChangeInformationImageFilter->GetChangeSpacing());
+    EXPECT_FALSE(newChangeInformationImageFilter->GetChangeOrigin());
+    EXPECT_FALSE(newChangeInformationImageFilter->GetChangeDirection());
+    EXPECT_FALSE(newChangeInformationImageFilter->GetChangeRegion());
+    EXPECT_FALSE(newChangeInformationImageFilter->GetCenterImage());
+    EXPECT_FALSE(newChangeInformationImageFilter->GetUseReferenceImage());
+
+    EXPECT_TRUE(newChangeInformationImageFilter->GetOutputDirection().GetVnlMatrix().is_identity());
+
+    const auto expectAllEqualTo = [](const auto & range, const auto expectedValue) {
+      // Expects that all elements are equal to the expected value.
+      for (const auto element : range)
+      {
+        EXPECT_EQ(element, expectedValue);
```

Comment:
This is an interesting way to do vector comparisons with nested lambda functions. 

One down side to this approach is that the produced GTest message will not be helpful in tracing the failed comparison.

Does the existing ITK_EXPECT_VECTOR_NEAR work?
Please consider improving and reusing GTest Predicates in itkGTestPredicate.h.

Reply:
Interesting suggestion, @blowekamp Thanks!

The existing ITK_EXPECT_VECTOR_NEAR does not _exactly_ do the same as the little `expectAllEqualTo` lambda I'm proposing here. You see `expectAllEqualTo` compares each element to one specific expected element value. And it expects them to be _exactly_ equal. I agree with you that the produced GTest messages from `expectAllEqualTo` may be less clear, although it won't be "rocket science" to figure out what went wrong (in case of a failure). And in practice I don't "expect"  it to _ever_ fail anyway, for this particular test.

We could add an extra macro, like `ITK_EXPECT_ALL_EQUAL_TO(rangeOfValues, expectedValue)`. But for the moment, I think the current PR is good enough. Right? Then maybe extend "itkGTestPredicate.h" later...

Reply:
Yes, I agree that the current PR is fine.

I am just suggesting creating reusable gtest predicates, in stead of lost if little comparison code every where.

The ITK_EXPECT_VECTOR_NEAR could work good enough with a rms error of ~0, and a vector type constructed with the scalar for all values.